src/puppetlabs/puppetdb/http/nodes.clj

```diff
@@ -46,10 +47,12 @@
         (validate-query-params {}))}
```

Comment:
At a glance, I don't hate this idea. It would involve two queries to just re-use the existing pattern, unless we change the wrap-with-parent-check to deal with using a different testing methodology, one that re-uses the query. Let me ponder.